components/PyTorch/pytorch-kfp-components/pytorch_kfp_components/components/base/base_component.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env/python3
+#
+# Copyright (c) Facebook, Inc. and its affiliates.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Pipeline Base component class."""
+
+import abc
+from six import with_metaclass
+from pytorch_kfp_components.types import standard_component_specs
+
+
+class BaseComponent(with_metaclass(abc.ABCMeta, object)):  # pylint: disable=R0903
```

Comment:
What is the main purpose of this class?
What users are supposed to use this class and how?